parsec/core/fs/userfs/userfs.py

```diff
@@ -0,0 +1,769 @@
+            FSBackendOfflineError
+        """
+        if not self.get_user_manifest().get_workspace_entry(workspace_id):
+            raise FSWorkspaceNotFoundError(f"Unknown workspace `{workspace_id}`")
+
+        try:
+            rights = await self.backend_cmds.vlob_group_get_rights(workspace_id)
+
+        except BackendNotAvailable as exc:
+            raise FSBackendOfflineError(str(exc)) from exc
+
+        except BackendCmdsNotAllowed:
+            # Seems we lost all the access rights
+            self_rights = {"admin_right": False, "read_right": False, "write_right": False}
+
+        except BackendConnectionError as exc:
+            raise FSError(f"Cannot retrieve workspace per-user rights: {exc}") from exc
+
+        else:
+            try:
+                self_rights = rights[self.device.user_id]
+
+            except KeyError:
+                # We should never be here in theory given if we had no rights
+                # backend was supposed to deny us the access !
+                logger.warning(
+                    "User not part of the rights returned by the backend",
+                    vlob_group=workspace_id,
+                    device_id=self.device.device_id,
+                )
+                self_rights = {"admin_right": False, "read_right": False, "write_right": False}
+
+        async with self._update_user_manifest_lock:
+            user_manifest = self.get_user_manifest()
+            workspace_entry = user_manifest.get_workspace_entry(workspace_id)
+            updated_workspace_entry = workspace_entry.evolve(**self_rights)
+            updated_user_manifest = user_manifest.evolve_workspaces_and_mark_updated(
+                updated_workspace_entry
+            )
+            self.local_storage.set_dirty_manifest(self.user_manifest_access, updated_user_manifest)
+
+    def get_workspace(self, workspace_id: UUID) -> WorkspaceFS:
+        """
+        Raises:
+            FSWorkspaceNotFoundError
+        """
+        user_manifest = self.get_user_manifest()
+        workspace_entry = user_manifest.get_workspace_entry(workspace_id)
+        if not workspace_entry:
+            raise FSWorkspaceNotFoundError(f"Unknown workspace `{workspace_id}`")
+
+        # TODO: WorkspaceFS shouldn't need workspace name (especially since it
+        # could be changed at anytime !), pass the workspace entry object instead
+        return WorkspaceFS(
+            workspace_entry.name,
+            device=self.device,
+            local_storage=self.local_storage,
+            backend_cmds=self.backend_cmds,
+            event_bus=self.event_bus,
+            _local_folder_fs=self._local_folder_fs,
+            _file_transactions=self._file_transactions,
+            _remote_loader=self._remote_loader,
+            _syncer=self._syncer,
+        )
+
+    async def workspace_create(self, name: str) -> UUID:
+        """
+        Raises: Nothing !
+        """
+        workspace_entry = WorkspaceEntry(name)
```

Comment:
Nope, workspace entry's name has no technical use (workspace id is used instead)
The idea is workspace entry' names can be fairly easily corrected by the user himself (and this save us sync conflict resolution).

Reply:
Ok, but what happens in the application when two names collide?

Reply:
nothing ! GUI shows two workspaces with same name (but with different content)

However I didn't test the case we mount two workspaces with same name (second mount should fail I guess, however this should be fixed for a broader usecase when we couldn't do the mount because the mounting folder is already in use)

Reply:
So mounting two workspaces with the same name is simply banned?